File: 05/CSharp/ExpressionApp/Divide.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExpressionApp
{
    class Divide : BinaryOperator
    {
        public Divide(IExpression first, IExpression second)
        {
            fOp = first;
            sOp = second;
        }

        public override double Evaluate(double param)
        {
            return fOp.Evaluate(param) / sOp.Evaluate(param);
        }
    }
}

File: 05/Java/expression/Divide.java
package Expression;

public class Divide extends BinaryOperator implements Expression {
    public Divide(Expression first, Expression second) {
        fOp = first;
        sOp = second;
    }

    public double evaluate(double param) {
        return fOp.evaluate(param) / sOp.evaluate(param);
    }
}